### PR TITLE
Add presto tls related resources.

### PR DESCRIPTION
**Pull request by MarvinCai**
### Motivation
Necessary changes to support tls for presto.

### Modifications
Generate and mount tls cert for presto.

### Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 
- [x] `no-need-doc`

### CODE REVIEW

File: charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml [hunk 1]
```diff
@@ -106,6 +106,15 @@ data:
     brokerClientTrustCertsFilePath: "/pulsar/certs/ca/ca.crt"
     {{- end }}
 
+    {{- if .Values.functions.skipTls }}
```
Comment (fantapsody) [inline charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml:109]: The name `skipTls` is a bit confusing, actually the traffic is still on TLS in this case, it just does not reject invalid certificates.

File: charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml [hunk 2]
```diff
@@ -106,6 +106,15 @@ data:
...
+    tlsAllowInsecureConnection: false
+    tlsEnableHostnameVerification: false
+    {{- end }}
+    
+    {{- if .Values.functions.RefreshCheckDurationSec }}
```
Comment (fantapsody) [inline charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml:114]: Should the name starts with a lowercase letter?

File: charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml [hunk 3]
```diff
@@ -129,6 +129,11 @@ spec:
           {{- end }}
           command: ["sh", "-c"]
           args:
+            {{ if .Values.presto.coordinator.commandOverride }}
+            {{ with .Values.presto.coordinator.commandOverride }}
+            -{{- toYaml . | indent 12 }}
+            {{- end }}
+            {{ else }}
```
Comment (fantapsody) [inline charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml:136]: Do we need to use `{{- else }}` and `{{- end }}` here to chomp whitespaces?

File: charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml [hunk 4]
```diff
@@ -202,6 +208,16 @@ spec:
               name: ca
               readOnly: true
             {{- end}}
+            {{- if and .Values.tls.enabled (or .Values.tls.broker.enabled (or .Values.tls.zookeeper.enabled .Values.tls.bookie.enabled)) }}
```
Comment (fantapsody) [inline charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml:211]: Does presto talk to bk and zk directly?

File: charts/pulsar/templates/presto/presto-worker-statefulset.yaml [hunk 5]
```diff
@@ -130,6 +130,11 @@ spec:
           {{- end }}
           command: ["sh", "-c"]
           args:
+            {{ if .Values.presto.worker.commandOverride }}
+            {{ with .Values.presto.worker.commandOverride }}
+            -{{- toYaml . | indent 12 }}
+            {{- end }}
+            {{ else }}
```
Comment (fantapsody) [inline charts/pulsar/templates/presto/presto-worker-statefulset.yaml:137]: The same whitespace control question as above.

File: charts/pulsar/templates/tls/tls-certs-internal.yaml [hunk 6]
```diff
@@ -66,6 +66,44 @@ spec:
 {{- end }}
 {{- end }}
 
+{{- if or .Values.tls.zookeeper.enabled .Values.tls.bookie.enabled }}
+apiVersion: cert-manager.io/v1alpha2
```
Comment (fantapsody) [inline charts/pulsar/templates/tls/tls-certs-internal.yaml:70]: Has this API version been deprecated in recent cert-manager versions?

File: charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml (hunk 1 above)
Comment (MarvinCai) [reply charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml:109]: updated to .Values.functions.tlsAllowInsecureConn

File: charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml (hunk 2 above)
Comment (MarvinCai) [reply charts/pulsar/templates/broker/function-worker-configfile-configmap.yaml:114]: updated to .Values.functions.tlsCertRefreshCheckDurationSec

File: charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml (hunk 3 above)
Comment (MarvinCai) [reply charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml:136]: done

File: charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml (hunk 4 above)
Comment (MarvinCai) [reply charts/pulsar/templates/presto/presto-coordinator-statefulset.yaml:211]: yes, presto get bk address from zk and actually directly read data from bookkeeper

File: charts/pulsar/templates/presto/presto-worker-statefulset.yaml (hunk 5 above)
Comment (MarvinCai) [reply charts/pulsar/templates/presto/presto-worker-statefulset.yaml:137]: done

File: charts/pulsar/templates/tls/tls-certs-internal.yaml (hunk 6 above)
Comment (MarvinCai) [reply charts/pulsar/templates/tls/tls-certs-internal.yaml:70]: that's right, but seems all cert in this template is in v1alpha2 version, will have a separate pr to upgrade to v1 version.

**Review by fantapsody — APPROVED** [review]

**Pull request merged**